scalafmt throws an ERROR on valid code with match pattern and if-guard

This template is a guideline, not a strict requirement.
- **Version**: >= 2.5.0
- **Integration**: sbt-scalafmt 2.4.0
- **Configuration**:
```
version = 2.7.0
style = defaultWithAlign
maxColumn = 120
encoding = UTF-8
docstrings = JavaDoc
assumeStandardLibraryStripMargin = true
spaces.inImportCurlyBraces = true

project.git = true

rewrite.rules = [
  RedundantBraces
  RedundantParens
  SortImports
  SortModifiers
  PreferCurlyFors
]
```

## Steps

Given code like this:
```scala
object Main extends App {
  
    val test: (Boolean, Boolean) = (false, true)

    val result = test._1 match {
        case true if (test._2 match {
            case true  => true
            case false => false
        }) => "so true"
        case false if (test._2 match {
            case true => true
            case false   => false
        }) => "still false"
        case _ => "so false"
    }

    Console.println(result)
}
```

When I run scalafmt like this:

```bash
scalafmt
```

## Problem

Scalafmt returns an error like this:
```sbt
ERROR scalafmt: Main.scala:8: error: => expected but match found
        case true if  test._2 match {
                              ^
```

## Expectation

I would like to not get an error and that the code is formatted

## Workaround

I've found that by reverting to at least scalafmt `version = 2.4.2` I do not get an error and the code is formatted.
I start to get the error with `version = 2.5.0`

## Notes
`scalaVersion := "2.13.3"`
`sbt.version=1.3.13`

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala
@@ -28,9 +28,9 @@ class RedundantParens(implicit ctx: RewriteCtx) extends RewriteSession {
   private[rewrite] val rewriteFunc: PartialFunction[Tree, Unit] = {
     case t @ (_: Term.Tuple | _: Type.Tuple | _: Lit.Unit) => remove(t, 2)
 
-    case g: Enumerator.Guard => remove(g.cond)
+    case g: Enumerator.Guard => maybeRemovePostfix(g.cond)
 
-    case t: Case => t.cond.foreach(remove(_))
+    case t: Case => t.cond.foreach(maybeRemovePostfix)
 
     case t if usesAvoidInfix && t.parent.exists {
           case p: Term.ApplyInfix => p.lhs ne t
@@ -73,6 +73,15 @@ class RedundantParens(implicit ctx: RewriteCtx) extends RewriteSession {
       case _ if rewriteFunc.isDefinedAt(tree) =>
       case InfixApp(ia) if breaksBeforeOp(ia) => // can't rewrite
       case _ => remove(tree, minToKeep)
+    }
+
+  // https://www.scala-lang.org/files/archive/spec/2.13/06-expressions.html#prefix-infix-and-postfix-operations
+  private def maybeRemovePostfix(tree: Tree): Unit =
+    tree match {
+      case _: Lit | _: Term.Name | _: Term.ApplyInfix | _: Term.Select |
+          _: Term.Apply | _: Term.ApplyUnary =>
+        remove(tree)
+      case _ =>
     }
 
   private def remove(tree: Tree, minToKeep: Int = 0): Unit =

--- a/scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat
+++ b/scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat
@@ -154,10 +154,14 @@ rewrite.rules = [AvoidInfix, RedundantParens]
 ===
 object a {
   val z = (((insertData) op (readDatabase(id))))
+  val z = (((insertData)
+   op (readDatabase(id))))
 }
 >>>
 object a {
   val z = insertData.op(readDatabase(id))
+  val z = insertData
+    .op(readDatabase(id))
 }
 <<< #1895 4
 object a {
@@ -301,3 +305,101 @@ val map = (Map("A" -> 1) - "Test" - "Test" - "Test" - "Test" - "Test"
 >>>
 val map = (Map("A" -> 1) - "Test" - "Test" - "Test" - "Test" - "Test"
   - "Test" - "Test" - "Test" - "Test" - "Test" - "Test" - "Test" + "B" -> 2)
+<<< #2194 1 match
+object a {
+  val b = c match {
+    case true if (d match {
+      case true  => true
+      case false => false
+    }) =>
+  }
+}
+>>>
+test does not parse
+object a {
+  val b = c match {
+    case true if (d match {
+                  ^
+<<< #2194 2 infix
+object a {
+  val b = c match {
+    case true if (d map {
+      case true  => true
+      case false => false
+    }) =>
+    case true if (d
+      map {
+        case true  => true
+        case false => false
+      }) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case true if d map {
+          case true  => true
+          case false => false
+        } =>
+    case true
+        if d
+          map {
+            case true  => true
+            case false => false
+          } =>
+  }
+}
+<<< #2194 3 apply with brace
+object a {
+  val b = c match {
+    case true if (d.map {
+      case true  => true
+      case false => false
+    }) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case true if d.map {
+          case true  => true
+          case false => false
+        } =>
+  }
+}
+<<< #2194 4 apply with paren
+object a {
+  val b = c match {
+    case true if (d.map({
+      case true  => true
+      case false => false
+    })) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case true if d.map({
+          case true  => true
+          case false => false
+        }) =>
+  }
+}
+<<< #2194 5 unary
+object a {
+  val b = c match {
+    case false if (+1) =>
+    case false if (-1) =>
+    case false if (~1) =>
+    case false if (!1) =>
+  }
+}
+>>>
+object a {
+  val b = c match {
+    case false if +1 =>
+    case false if -1 =>
+    case false if ~1 =>
+    case false if !1 =>
+  }
+}

--- a/scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat
+++ b/scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat
@@ -315,11 +315,14 @@ object a {
   }
 }
 >>>
-test does not parse
 object a {
   val b = c match {
     case true if (d match {
-                  ^
+          case true  => true
+          case false => false
+        }) =>
+  }
+}
 <<< #2194 2 infix
 object a {
   val b = c match {

